### ISSUE TITLE
Potential bug during load 0.7.0-beta2 on ng-include (complex example)

### ISSUE THREAD

**Issue by deenairn** [handsontable/ngHandsontable#118]
I've got an issue where the I have a relatively complex screen with a factory and an ng-include where on the second load of the data, the data is not loaded correctly. Unfortunately I cannot share the example due to copyright reasons, and I have been unable to exactly track down the source of the issue. The code is very complicated and trying to narrow down into a jsFiddle or Plunker example is not very easy.

However, I noticed that if I change lines below (these were changed in 0.7.0-beta2 from 0.7.0-beta1 version, it previously worked under 0.7.0-beta1):

```
          // TODO: Add watch properties descriptor + needs perf test. Watch full equality vs toJson
          angular.forEach(bindingsKeys, function(key) {
            scope.$watch(key, function(newValue, oldValue) {
              if (newValue === void 0 || newValue === oldValue)) {
                return;
              }
              if (key === 'datarows') {
                // If reference to data rows is not changed then only re-render table
                if (scope.hotInstance.getSettings().data === newValue) {
                  settingFactory.renderHandsontable(scope.hotInstance);
                } else {
                  scope.hotInstance.loadData(newValue);
                }
              } else {
                scope.htSettings[key] = newValue;
                settingFactory.updateHandsontableSettings(scope.hotInstance, scope.htSettings);
              }
            }, ['datarows', 'columns', 'rowHeights', 'colWidths', 'rowHeaders', 'colHeaders'].indexOf(key) >= 0);
          });
```

to

```
          // TODO: Add watch properties descriptor + needs perf test. Watch full equality vs toJson
          angular.forEach(bindingsKeys, function(key) {
            scope.$watch(key, function(newValue, oldValue) {
              if (newValue === void 0) {
                return;
              }
              if (key === 'datarows') {
                // If reference to data rows is not changed then only re-render table
                if (scope.hotInstance.getSettings().data === newValue) {
                  settingFactory.renderHandsontable(scope.hotInstance);
                } else {
                  scope.hotInstance.loadData(newValue);
                }
              } else if (newValue === oldValue) {
                scope.htSettings[key] = newValue;
                settingFactory.updateHandsontableSettings(scope.hotInstance, scope.htSettings);
              }
            }, ['datarows', 'columns', 'rowHeights', 'colWidths', 'rowHeaders', 'colHeaders'].indexOf(key) >= 0);
          });
```

The table then appears correctly. I don't believe the second implementation loses any of the functionality of the first and may fix the problem I have (which may appear elsewhere??).

The only alternative I have is to force a digest cycle using $apply ($timeout does not work) which gives a digest cycle already in progress.

Do you have any idea why this might be happening?

In addition, I noticed the following in the codebase:

```
            if (!$rootScope.$$phase) {
              scope.$apply();
            }
```

This is going to be phased out and the $timeout service is apparently the new way to do this.

http://stackoverflow.com/questions/12729122/angularjs-prevent-error-digest-already-in-progress-when-calling-scope-apply

**Comment by deenairn** [comment]
The second code block should be - I got the conditional wrong at the end. I tried changing this on the repo in hotTable.js and the jasmine tests still pass.
```
          // TODO: Add watch properties descriptor + needs perf test. Watch full equality vs toJson
          angular.forEach(bindingsKeys, function(key) {
            scope.$watch(key, function(newValue, oldValue) {
              if (newValue === void 0) {
                return;
              }
              if (key === 'datarows') {
                // If reference to data rows is not changed then only re-render table
                if (scope.hotInstance.getSettings().data === newValue) {
                  settingFactory.renderHandsontable(scope.hotInstance);
                } else {
                  scope.hotInstance.loadData(newValue);
                }
              } else if (newValue !== oldValue) {
                scope.htSettings[key] = newValue;
                settingFactory.updateHandsontableSettings(scope.hotInstance, scope.htSettings);
              }
            }, ['datarows', 'columns', 'rowHeights', 'colWidths', 'rowHeaders', 'colHeaders'].indexOf(key) >= 0);
          });
```

**Comment by budnix** [comment]
Hi, thanks for support. Your changes was included in 0.7.0-beta3.
